### PR TITLE
Issue#42 Write an ADR on our choices of database structure

### PR DESCRIPTION
**Pull request by JolanThomassin**
Name: **dev-rel-docs/adr/002-database-structure.md**

This ADR explains our plan to **organize our databases better**. You can find all the details in this [document ]( https://github.com/ai-cfia/louis-db/issues/24). In a nutshell, we're restructuring our databases to work together smoothly as we expand our product range. We'll have a **main repository named "ailab-db"** and add modules for specific products like "louis," "nachet," or "finesse." This approach offers flexibility and adaptability. The ADR aims to **clarify how** we're organizing everything **and why** we're doing it this way.

### CODE REVIEW

**Comment by JolanThomassin** [comment]
_This comment will be moved in the Pull Request when created_

While it may not be perfect and might be too much focus on the transition when the document need to be focus on the final state, the structure can resemble the following:

# Title:
ADR on our choices of database structure

# Status:
Proposed

# Evaluation Criteria:
We aim to determine the most effective database structure that allows for seamless integration of multiple products, improves flexibility, maintainability, and fosters collaboration among development teams.

# Candidates to Consider:
We have considered various database structure options to accommodate multiple products, including maintaining the existing separate databases and our proposed restructuring approach.

# Research and Analysis of Each Candidate:
In our analysis, we thoroughly examined the existing "louis-db" and "nachet-db" structures and assessed the proposed restructuring. We found that the current isolated databases limited adaptability, while the proposed integrated approach promised improved efficiency and collaboration, similar to transitioning from separate drawers to a spacious wardrobe. The proposed changes would streamline maintenance, reduce errors, and support our growing product variety, making them the preferred choice for our database structure.

# Does/Doesn't Meet Criteria and Why:
The existing separate database approach doesn't meet our criteria as it leads to duplication, lacks flexibility, and hinders collaboration. The proposed changes meet our criteria as they streamline integration, improve maintainability, and enhance code quality.

# Internal Opinions and Feedback:
_Waiting for team members opinion about the proposed changes._

# External Opinions and Feedback:
_Waiting for external opinion about the proposed changes._

# How Did You Evaluate the Candidates?
We evaluated the candidates based on the criteria of efficiency, flexibility, maintainability, and collaborative potential. The proposed changes best align with these criteria.

Why Did You Choose the Winner?
We chose the proposed changes because they align with our objective of creating a single, easy-to-maintain database system that can seamlessly integrate multiple products.

What Is Happening Since Then?
The proposed changes are under consideration for implementation, with a focus on developing the "ailab-db" as the core database and creating a modular product integration system.

Knowing What You Know Now, What Would You Advise People to Do Differently?
We would advise others to consider the long-term benefits of restructuring their database systems to accommodate growth and product diversity, even if it requires initial changes and investments.

Recommendation:
We recommend implementing the proposed changes, including renaming the "louis-db" to "ailab-db" and revising the repository structure for seamless product integration. These changes align with our objective and are vital for efficiently managing databases for multiple products, streamlining deployment and maintenance processes, improving code quality, and facilitating collaboration among development teams.

**Review by rngadam — CHANGES_REQUESTED** [review]
starting point should be the diagram you created and an explanation of the diagram.

do not use meaningless consulting buzzwords.

File: adr/002-database-structure.md [hunk 1]
```diff
@@ -1,17 +1,20 @@
+![AiLab-db](https://github.com/ai-cfia/dev-rel-docs/assets/98430140/d1795d78-8a13-4c94-86bb-94dd8f824581)
```
Comment (rngadam) [inline adr/002-database-structure.md:1]: don't use absolute links. these will break if we move things around

File: adr/002-database-structure.md [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
 
 This ADR details our decisions regarding the database structure and highlights the factors influencing our choices.
 
 ## Top-level
 - **Title:** ADR on our choices of database structure
 - **Status:** Proposed
-- **Evaluation Criteria:** We aim to determine the most effective database structure that allows for seamless integration of multiple products, improves flexibility, maintainability, and fosters collaboration among development teams.
-- **Candidates to Consider:** We have considered various database structure options to accommodate multiple products, including maintaining the existing separate databases and our proposed restructuring approach.
+- **Evaluation Criteria:** We want to figure out the best way to set up the database so that many products can work together smoothly. This will make things flexible, easy to take care of, and help different teams work together better.
+- **Candidates to Consider:** We looked at different ways to set up the database for lots of products, like keeping them separate or changing how they're organized.
```
Comment (rngadam) [inline adr/002-database-structure.md:11]: that's not concrete enough, you need to explain what these alternatives are

File: adr/002-database-structure.md [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
 
 ## Research and Analysis of Each Candidate
-In our analysis, we thoroughly examined the existing "louis-db" and "nachet-db" structures and assessed the proposed restructuring. We found that the current isolated databases limited adaptability, while the proposed integrated approach promised improved efficiency and collaboration, similar to transitioning from separate drawers to a spacious wardrobe. The proposed changes would streamline maintenance, reduce errors, and support our growing product variety, making them the preferred choice for our database structure.
+In our investigation, we looked very closely at how "louis-db" and "nachet-db" are set up right now, and we also checked out the idea to change things. We found that having separate databases like drawers made it hard to change and work together. But if we put everything in one big "wardrobe," it would be much better. This change would make it easier to take care of, prevent mistakes, and help us with all the different products we have. So, we think switching to the big "wardrobe" is the best way to go for our database.
```
Comment (rngadam) [inline adr/002-database-structure.md:14]: this document will be given to dba and dbdev, I don't think we want analogies like drawers and wardrobe

Comment (rngadam) [reply adr/002-database-structure.md:14]: drop the "we think" from your sentences.

https://hbr.org/2016/10/stop-trying-to-sound-smart-when-youre-writing
https://medium.com/the-brave-writer/stop-using-this-phrase-when-you-write-4dd18c81de49

File: adr/002-database-structure.md [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
 
-- **Does/Doesn't Meet Criteria and Why:** The existing separate database approach doesn't meet our criteria as it leads to duplication, lacks flexibility, and hinders collaboration. The proposed changes meet our criteria as they streamline integration, improve maintainability, and enhance code quality.
+- **Does/Doesn't Meet Criteria and Why:**
+The way we have separate databases right now isn't good because it causes repetition, is not flexible, and makes it hard for us to work together. But the changes we want to make are better because they make things work together, are easier to manage, and make the computer code better.
```
Comment (rngadam) [inline adr/002-database-structure.md:17]: not concrete

File: adr/002-database-structure.md [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
 
 ## Opinions and Feedback
 - **Internal Opinions and Feedback:** Waiting for team members' opinions about the proposed changes.
```
Comment (rngadam) [inline adr/002-database-structure.md:20]: don't put solicitation for opinions in the document itself. that's why we have a PR open

File: adr/002-database-structure.md [hunk 6]
```diff
@@ -21,12 +24,13 @@ In our analysis, we thoroughly examined the existing "louis-db" and "nachet-db"
...
+We think it's a good idea for others to think about how changing their database now can help them in the future when they have more products and need to grow. Sometimes, making some changes and spending some money at the beginning can be really helpful later on.
 
-- **Recommendation:** We recommend implementing the proposed changes, including renaming the "louis-db" to "ailab-db" and revising the repository structure for seamless product integration. These changes align with our objective and are vital for efficiently managing databases for multiple products, streamlining deployment and maintenance processes, improving code quality, and facilitating collaboration among development teams.
+- **Recommendation:**
+We suggest making the changes we talked about, like calling "louis-db" by a new name, "ailab-db," and arranging things in a better way. These changes match our goal and are super important for handling lots of products, making things easier to use, taking care of the computer code, and helping teams work together better.
```
Comment (rngadam) [inline adr/002-database-structure.md:36]: you can't get more vague than "arranging THINGS in a better WAY"...

**Review by rngadam — CHANGES_REQUESTED** [review]

File: adr/002-database-structure.md (hunk 5 above)
Comment (JolanThomassin) [reply adr/002-database-structure.md]: Done

File: adr/002-database-structure.md (hunk 1 above)
Comment (JolanThomassin) [reply adr/002-database-structure.md]: Done

File: adr/002-database-structure.md (hunk 2 above)
Comment (JolanThomassin) [reply adr/002-database-structure.md]: Done

File: adr/002-database-structure.md (hunk 3 above)
Comment (JolanThomassin) [reply adr/002-database-structure.md]: Done

File: adr/002-database-structure.md (hunk 6 above)
Comment (JolanThomassin) [reply adr/002-database-structure.md]: Done

File: adr/002-database-structure.md (hunk 4 above)
Comment (JolanThomassin) [reply adr/002-database-structure.md]: I changed it, Done

File: adr/002-database-structure.md (hunk 3 above)
Comment (rngadam) [reply adr/002-database-structure.md]: I still see drawers and wardrobe in the latest version.

**Comment by rngadam** [comment]
@JolanThomassin You've marked comments as Done but when I look at the latest version in Github, it's still not done for many of them. Did you mean to flag this as Draft? Have you pushed your changes?

Feels like you're writing as if you were getting paid by the word:

_"We think it's a good idea "_ 

_"The proposed changes are under consideration for implementation, with a focus"_

_"We picked the changes we suggested because"_

**Comment by JolanThomassin** [comment]
@rngadam In another PR, I was commenting like this, but you told me to comment "Done," so I misunderstood and thought I was only supposed to write "Done" when it was actually done.

As for the missing fix, this probably happened because I messed something up in the PR system and created two branches. You may have looked at the wrong one. In [this branch](https://github.com/ai-cfia/dev-rel-docs/blob/Issue%2342-Write-an-ADR-on-our-choices-of-database-structure/adr/002-database-structure.md), you can see all the desired changes. You can also find them in the last commits:

image

**Comment by rngadam** [comment]
@JolanThomassin peux tu simplifier ton PR et ne pas empiler deux PR une par dessus l'autre pour un ADR? je comprends pas pourquoi faire ça complexe pour un texte.

**Comment by rngadam** [comment]
I'm being asked to re-review but I don't see significant improvements here.

**Comment by JolanThomassin** [comment]
Merge because of branch mistake

**Pull request merged**